### PR TITLE
[Android] Add debug for android scripts

### PR DESCRIPTION
**Pull request by talregev**
Add debug for android scripts
Environment variable BOINC_DEBUG=ON
Add CFLAGS and CXXFLAGS can added from outside.

### CODE REVIEW

**Comment by adamradocz** [comment]
Using only `DEBUG` instead of `BOINC_DEBUG` wouldn't be a better choice?

**Comment by talregev** [comment]
I am not sure, Deubg is so general, so maybe it will do side effects for things we  don't know.

File: android/build_boinc_arm.sh [hunk 1]
```diff
@@ -25,8 +25,8 @@ export PATH="$TCBINARIES:$TCINCLUDES/bin:$PATH"
 export CC=arm-linux-androideabi-clang
 export CXX=arm-linux-androideabi-clang++
 export LD=arm-linux-androideabi-ld
-export CFLAGS="--sysroot=$TCSYSROOT -DANDROID -DDECLARE_TIMEZONE -Wall -I$TCINCLUDES/include -O3 -fomit-frame-pointer -fPIE -march=armv7-a -D__ANDROID_API__=19"
-export CXXFLAGS="--sysroot=$TCSYSROOT -DANDROID -Wall -I$TCINCLUDES/include -funroll-loops -fexceptions -O3 -fomit-frame-pointer -fPIE -march=armv7-a -D__ANDROID_API__=19"
+export CFLAGS="--sysroot=$TCSYSROOT -DANDROID -DDECLARE_TIMEZONE -Wall -I$TCINCLUDES/include -O3 -fomit-frame-pointer -fPIE -march=armv7-a -D__ANDROID_API__=19 "$CFLAGS
```
Comment (AenBleidd) [inline android/build_boinc_arm.sh:28]: This change adds no support of Debug build. You can leave here an ability to add parameters from the outside but also please override this value with `-g` parameter for Debug builds.

File: android/build_boinc_arm.sh [hunk 2]
```diff
@@ -51,7 +51,9 @@ if [ -n "$COMPILEBOINC" ]; then
 
     echo "Stripping Binaries"
     cd stage/usr/local/bin
-    arm-linux-androideabi-strip *
+    if [ ! "$BOINC_DEBUG" = "ON" ]; then
```
Comment (AenBleidd) [inline android/build_boinc_arm.sh:54]: Let's use general `DEBUG`. I see no reason why it should have some side effects.

File: android/build_boinc_arm.sh (hunk 1 above)
Comment (talregev) [reply android/build_boinc_arm.sh:28]: I am not sure I understand

**Comment by talregev** [comment]
> These two comments are applicable to all other scripts.
> Also, if we are building Debug, it should be better to build openssl and curl in Debug mode too.
> So please adjust corresponding scripts too.

The openssl and curl already build in debug mode.

**Comment by talregev** [comment]
From my test all the binaries are build in debug mode. Boinc, openssl, and curl.
It strip off the debug symbol by strip command.
When the Debug flag is on, it not doing the strip and it stay at debug.
I am not passing the -g throw outside CFLAGS and CXXFLGAS because the -g flag already in GDB_CFLAGS in all scripts (boinc, openssl curl) in all platforms.

**Comment by talregev** [comment]
So if you allow to add external CFLAGS and CXXFLAGS in this PR,
All I need to do is to change BOINC_DEBUG to DEBUG.

File: android/build_boinc_arm.sh (hunk 1 above)
Comment (AenBleidd) [reply android/build_boinc_arm.sh:28]: Apply then ability to add parameters from the outside to all other scripts

**Comment by truboxl** [comment]
I think GDB_CFLAGS was not used at all, if ever
So I think can remove them and directly integrate them into CFLAGS and/or CXXFLAGS based on DEBUG script flag

Please test if it ever works with LLDB, I can't seem to get it work

**Comment by talregev** [comment]
I compile the boinc with GDB_CFLGAS empty and without the strip and I get a debug information.
I don't know how to compile in release mode without the strip.
@AenBleidd  What do you want to do about this?

**Pull request closed without merging**